### PR TITLE
Add one-line descriptions to 19 examples currently missing them

### PR DESCRIPTION
**Pull request by tonyyli**
## PR Summary

I've added one-line descriptions to 19 examples that currently do not have them.

This partially addresses one of the issues mentioned in #8885 ("Ensure that all examples / tutorials have a title and a one-line description underneath").

@choldgraf @NelleV

### CODE REVIEW

File: examples/pylab_examples/axes_demo.py [hunk 1]
```diff
@@ -3,6 +3,7 @@
 Axes Demo
 =========
 
+Example use of `plt.axes` to create inset axes within the main plot axes.
```
Comment (choldgraf) [inline examples/pylab_examples/axes_demo.py]: use `` rather than ` when showing code style in rST

File: examples/pylab_examples/axhspan_demo.py [hunk 2]
```diff
@@ -3,6 +3,7 @@
 Axhspan Demo
 ============
 
+The example shows how to create lines or rectangles that span the axes in either the horizontal or vertical direction.
```
Comment (choldgraf) [inline examples/pylab_examples/axhspan_demo.py]: ~~The example shows how to ~~

File: examples/pylab_examples/axis_equal_demo.py [hunk 3]
```diff
@@ -3,6 +3,7 @@
 Axis Equal Demo
 ===============
 
+This example how to set and adjust plots with equal axis ratios.
```
Comment (choldgraf) [inline examples/pylab_examples/axis_equal_demo.py]: ~~This example~~

File: examples/pylab_examples/contourf_demo.py [hunk 4]
```diff
@@ -3,6 +3,7 @@
 Contourf Demo
 =============
 
+Example use of the `contourf` function to create filled contour plots.
```
Comment (choldgraf) [inline examples/pylab_examples/contourf_demo.py]: ~~Example use of~~ How to use

File: examples/pylab_examples/contourf_hatching.py [hunk 5]
```diff
@@ -3,6 +3,7 @@
 Contourf Hatching
 =================
 
+Demo of filled contour plots with of hatched patterns.
```
Comment (choldgraf) [inline examples/pylab_examples/contourf_hatching.py]: ~~of~~

File: examples/pylab_examples/contourf_hatching.py [hunk 6]
```diff
@@ -29,6 +30,7 @@
                   )
 plt.colorbar()
 
+###############################################################################
```
Comment (choldgraf) [inline examples/pylab_examples/contourf_hatching.py]: let's remove this un lieu of the next PR

File: examples/pylab_examples/figure_title.py [hunk 7]
```diff
@@ -3,6 +3,7 @@
 Figure Title
 ============
 
+Example use of `title` and `suptitle` to create subplot titles and a centered figure title, respectively.
```
Comment (choldgraf) [inline examples/pylab_examples/figure_title.py]: `` not `

also, make sure this doesn't go over 79 characters...not sure if it does or not but it feels a little long to me :-)

File: examples/pylab_examples/fill_betweenx_demo.py [hunk 8]
```diff
@@ -3,6 +3,7 @@
 Fill Betweenx Demo
 ==================
 
+This example shows how to use `fill_betweenx` to color between two horizontal curves.
```
Comment (choldgraf) [inline examples/pylab_examples/fill_betweenx_demo.py]: ~~This example shows how to use~~ Using

also `` not `

File: examples/pylab_examples/plotfile_demo.py [hunk 9]
```diff
@@ -3,6 +3,7 @@
 Plotfile Demo
 =============
 
+Example use of `plotfile` to plot data directly from a file.
```
Comment (choldgraf) [inline examples/pylab_examples/plotfile_demo.py]: `` not `

File: examples/pylab_examples/simple_plot.py [hunk 10]
```diff
@@ -3,6 +3,7 @@
 Simple Plot
 ===========
 
+Simple example of a simple plot.
```
Comment (choldgraf) [inline examples/pylab_examples/simple_plot.py]: ~~Simple example of~~ Create

File: examples/pylab_examples/specgram_demo.py [hunk 11]
```diff
@@ -3,6 +3,7 @@
 Specgram Demo
```
Comment (choldgraf) [inline examples/pylab_examples/specgram_demo.py]: Let's call this `Spectrogram Demo`

File: examples/pylab_examples/symlog_demo.py [hunk 12]
```diff
@@ -3,6 +3,7 @@
 Symlog Demo
 ===========
 
+Example use of symlog (symmetrical log) axis scaling.
```
Comment (choldgraf) [inline examples/pylab_examples/symlog_demo.py]: `symmetric`? not sure if I'm right there or not...probably fine either way

File: examples/pylab_examples/xcorr_demo.py [hunk 13]
```diff
@@ -3,6 +3,7 @@
 Xcorr Demo
 ==========
 
+Example use of cross-correlation (`xcorr`) and auto-correlation (`acorr`) plots.
```
Comment (choldgraf) [inline examples/pylab_examples/xcorr_demo.py]: `` not `

File: examples/pylab_examples/xcorr_demo.py [hunk 14]
```diff
@@ -3,6 +3,7 @@
 Xcorr Demo
```
Comment (choldgraf) [inline examples/pylab_examples/xcorr_demo.py]: Cross-correlation Demo

File: examples/pylab_examples/symlog_demo.py (hunk 12 above)
Comment (tonyyli) [inline examples/pylab_examples/symlog_demo.py]: Seems ok. It's shorter anyway :)

**Comment by tonyyli** [comment]
Pushed some edits with the suggested changes

File: examples/pylab_examples/xcorr_demo.py [hunk 15]
```diff
@@ -1,8 +1,9 @@
 """
 ==========
-Xcorr Demo
+Cross-correlation Demo
```
Comment (tacaswell) [inline examples/pylab_examples/xcorr_demo.py]: The title lines on this one need to be the same length as the title (which is what is failing travis)

**Comment by tonyyli** [comment]
@choldgraf Conficts addressed, finished with current changes

File: examples/images_contours_and_fields/contourf_hatching.py [hunk 16]
```diff
@@ -3,6 +3,7 @@
 Contourf Hatching
 =================
 
+Demo filled contour plots with of hatched patterns.
```
Comment (choldgraf) [inline examples/images_contours_and_fields/contourf_hatching.py]: ~~of~~

File: examples/pylab_examples/spine_placement_demo.py [hunk 17]
```diff
@@ -3,6 +3,7 @@
 Spine Placement Demo
 ====================
 
+This example demos how to adjust the location and appearance of axis spines.
```
Comment (choldgraf) [inline examples/pylab_examples/spine_placement_demo.py]: ~~This example demos how to adjust~~ Adjusting

File: examples/pylab_examples/xcorr_acorr_demo.py [hunk 18]
```diff
@@ -1,8 +1,9 @@
 """
-==========
-Xcorr Demo
-==========
+============================
+Cross-/Auto-correlation Demo
```
Comment (choldgraf) [inline examples/pylab_examples/xcorr_acorr_demo.py]: how about "Cross- and Auto-Correlation Demo"?

**Comment by choldgraf** [comment]
A few small comments and after this LGTM

**Comment by tonyyli** [comment]
done.

**Comment by tacaswell** [comment]
@tonyyli Thanks and congratulations on your first Matplotlib contribution :tada:

**Comment by choldgraf** [comment]
Way to go @tonyyli ! And thanks for being patient during the review :-)

**Pull request merged**